core/src/main/java/io/smallrye/openapi/spi/OASFactoryResolverImpl.java

```diff
@@ -78,43 +78,48 @@
+            registry.put(Info.class, InfoImpl::new);
+            registry.put(License.class, LicenseImpl::new);
+            registry.put(Link.class, LinkImpl::new);
+            registry.put(MediaType.class, MediaTypeImpl::new);
+            registry.put(OAuthFlow.class, OAuthFlowImpl::new);
+            registry.put(OAuthFlows.class, OAuthFlowsImpl::new);
+            registry.put(OpenAPI.class, OpenAPIImpl::new);
+            registry.put(Operation.class, OperationImpl::new);
+            registry.put(Parameter.class, ParameterImpl::new);
+            registry.put(PathItem.class, PathItemImpl::new);
+            registry.put(Paths.class, PathsImpl::new);
+            registry.put(RequestBody.class, RequestBodyImpl::new);
+            registry.put(Schema.class, SchemaImpl::new);
+            registry.put(SecurityRequirement.class, SecurityRequirementImpl::new);
+            registry.put(SecurityScheme.class, SecuritySchemeImpl::new);
+            registry.put(Server.class, ServerImpl::new);
+            registry.put(ServerVariable.class, ServerVariableImpl::new);
+            registry.put(Tag.class, TagImpl::new);
+            registry.put(XML.class, XMLImpl::new);
+            REGISTRY = registry;
+        }
     }
 
-    static {
-        put(APIResponse.class, APIResponseImpl::new);
-        put(APIResponses.class, APIResponsesImpl::new);
-        put(Callback.class, CallbackImpl::new);
-        put(Components.class, ComponentsImpl::new);
-        put(Contact.class, ContactImpl::new);
-        put(Content.class, ContentImpl::new);
-        put(Discriminator.class, DiscriminatorImpl::new);
-        put(Encoding.class, EncodingImpl::new);
-        put(Example.class, ExampleImpl::new);
-        put(ExternalDocumentation.class, ExternalDocumentationImpl::new);
-        put(Header.class, HeaderImpl::new);
-        put(Info.class, InfoImpl::new);
-        put(License.class, LicenseImpl::new);
-        put(Link.class, LinkImpl::new);
-        put(MediaType.class, MediaTypeImpl::new);
-        put(OAuthFlow.class, OAuthFlowImpl::new);
-        put(OAuthFlows.class, OAuthFlowsImpl::new);
-        put(OpenAPI.class, OpenAPIImpl::new);
-        put(Operation.class, OperationImpl::new);
-        put(Parameter.class, ParameterImpl::new);
-        put(PathItem.class, PathItemImpl::new);
-        put(Paths.class, PathsImpl::new);
-        put(RequestBody.class, RequestBodyImpl::new);
-        put(Schema.class, SchemaImpl::new);
-        put(SecurityRequirement.class, SecurityRequirementImpl::new);
-        put(SecurityScheme.class, SecuritySchemeImpl::new);
-        put(Server.class, ServerImpl::new);
-        put(ServerVariable.class, ServerVariableImpl::new);
-        put(Tag.class, TagImpl::new);
-        put(XML.class, XMLImpl::new);
+    static <K extends Constructible, V extends K> void put(Class<K> key, Supplier<V> value) {
+        OASFactoryResolverRegistry.REGISTRY.put(key, value);
     }
 
```

Comment:
With your changes this put method can be removed also.

```suggestion
```

Reply:
wait a sec, so the `REGISTRY` cannot be modified?

Reply:
Yes, that's right.

Reply:
Does that change your thinking around the concurrency protection you mentioned earlier?